File: TransCoder_tokenized_test_set_functions/cpp/PARTITION_NUMBER_TWO_DIVISBLE_PARTS.cpp
void findDivision ( string & str, int a, int b ) {
  int len = str . length ( );
  vector < int > lr ( len + 1, 0 );
  lr [ 0 ] = ( str [ 0 ] - '0' ) % a;
  for ( int i = 1;
  i < len;
  i ++ ) lr [ i ] = ( ( lr [ i - 1 ] * 10 ) % a + ( str [ i ] - '0' ) ) % a;
  vector < int > rl ( len + 1, 0 );
  rl [ len - 1 ] = ( str [ len - 1 ] - '0' ) % b;
  int power10 = 10;
  for ( int i = len - 2;
  i >= 0;
  i -- ) {
    rl [ i ] = ( rl [ i + 1 ] + ( str [ i ] - '0' ) * power10 ) % b;
    power10 = ( power10 * 10 ) % b;
  }
  for ( int i = 0;
  i < len - 1;
  i ++ ) {
    if ( lr [ i ] != 0 ) continue;
    if ( rl [ i + 1 ] == 0 ) {
      cout << "YES\n";
      for ( int k = 0;
      k <= i;
      k ++ ) cout << str [ k ];
      cout << ", ";
      for ( int k = i + 1;
      k < len;
      k ++ ) cout << str [ k ];
      return;
    }
  }
  cout << "NO\n";
}

File: TransCoder_tokenized_test_set_functions/python/PARTITION_NUMBER_TWO_DIVISBLE_PARTS.py
def findDivision ( str, a, b ) :
    lenn = len ( str )
    lr = [ 0 ] * ( lenn + 1 )
    lr [ 0 ] = ( int ( str [ 0 ] ) ) % a
    for i in range ( 1, lenn ) :
        lr [ i ] = ( ( lr [ i - 1 ] * 10 ) % a + \ int ( str [ i ] ) ) % a
    rl = [ 0 ] * ( lenn + 1 )
    rl [ lenn - 1 ] = int ( str [ lenn - 1 ] ) % b
    power10 = 10
    for i in range ( lenn - 2, - 1, - 1 ) :
        rl [ i ] = ( rl [ i + 1 ] + int ( str [ i ] ) * power10 ) % b
        power10 = ( power10 * 10 ) % b
    for i in range ( 0, lenn - 1 ) :
        if ( lr [ i ] != 0 ) :
            continue
        if ( rl [ i + 1 ] == 0 ) :
            print ( "YES" )
            for k in range ( 0, i + 1 ) :
                print ( str [ k ], end = "" )
            print ( ",", end = " " )
            for i in range ( i + 1, lenn ) :
                print ( str [ k ], end = "" )
                return
    print ( "NO" )
